client/apiversion_test.go

```diff
@@ -0,0 +1,164 @@
+package client_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	gc "gopkg.in/check.v1"
+)
+
+type versionHandler struct {
+	authBody string
+	port     string
+}
+
+type makeServiceURLTest struct {
+	serviceType string
+	version     string
+	parts       []string
+	success     bool
+	URL         string
+	err         string
+}
+
+func (s *localLiveSuite) makeServiceURLTests() []makeServiceURLTest {
+	return []makeServiceURLTest{
+		{
+			serviceType: "compute",
+			version:     "v2.1",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.0/foo/bar/",
+		},
+		{
+			serviceType: "compute",
+			version:     "v2",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/v2.1/foo/bar/",
+		},
+		{
+			serviceType: "object-store",
+			version:     "",
+			parts:       []string{"foo", "bar"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar",
+		},
+		{
+			serviceType: "object-store",
+			version:     "v2.0",
+			parts:       []string{"foo", "bar/"},
+			success:     true,
+			URL:         "http://localhost:%s/swift/v1/foo/bar/",
```

Comment:
object-store only has one endpoint. But that endpoint does have a version, "v1".
So we can just have the MakeServiceURL caller in goose (and Juju which calls goose) pass in "" or "v1" and the correct, expected result will be returned. If the caller passes in "v2", that should still error.That allows consistent and understandable behaviour of the API, and keeps the semantics consistent without the special case.